### ISSUE TITLE
Symfony 2.8.18 breaks tests

### ISSUE THREAD

**Issue by aimeos** [symfony/symfony#21931]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | yes
| RFC?             | no
| Symfony version  | 2.8.18

The symfony Symfony 2.8.18 release contains a test that requires phpunit 6.0 which only is only available for PHP 7. Therefore, executing the unit tests with still supported PHP 5.6 is impossible because of this error:
```
PHP Fatal error:  Class 'PHPUnit\Framework\TestCase' not found in /home/travis/build/aimeos/aimeos-symfony/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php on line 25
```
Full stack trace can be found at https://travis-ci.org/aimeos/aimeos-symfony/jobs/208945815

**Comment by stof** [comment]
it does not requires PHPUnit 6. It requires 4.8.35+ or 5.4+

**Comment by aimeos** [comment]
In the `composer.json` yes, you are right but that's not the point.

This statement is only valid in phpunit 6.0:
`use PHPUnit\Framework\TestCase;`

https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php#L14

Guess this was introduced by backporting a change from Symfony 3

**Comment by robfrawley** [comment]
No, it is valid in the required PHPUnit versions, where the namespaced variants have been backported in PHPUnit itself.

**Comment by aimeos** [comment]
Requiring phpunit 4.5.35+ makes things extremly difficult to test on Travis-ci as they only provide phpunit 4.5. Using a phpunit version installed by composer always results in another Fatal Error from Doctrine:
```
PHP Fatal error:  Uncaught exception 'InvalidArgumentException' with message 'A callable is expected in AnnotationRegistry::registerLoader().' in /home/travis/build/aimeos/aimeos-symfony/vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationRegistry.php:111
```

**Comment by stof** [comment]
@aimeos I suspect that you are using ``$loader = require_once '.../vendor/autoload.php'``. This will *not* give you the ClassLoader instance if the file was already loaded previously, but ``true``. Use ``require`` instead (the composer ``vendor/autoload.php`` is precisely designed to be compatible with multiple requiring *because* it returns the loader).

**Comment by robfrawley** [comment]
You can download PHPUnit `4.8` using a statement like Symfony uses in their `.travis.yml` like:
```
wget https://phar.phpunit.de/phpunit-4.8.phar
```
See the [PR handling this for Symfony](https://github.com/symfony/symfony/pull/21694/files), or the [PR on new conflict rules](https://github.com/symfony/symfony/pull/21684),.

**Comment by xabbuh** [comment]
@aimeos see https://github.com/aimeos/aimeos-symfony/blob/master/Tests/bootstrap.php#L18

**Comment by craigh** [comment]
I am having similar trouble with failing tests since upgrading from 2.8.17 -> 2.8.18

2.18.17 -> [tests pass](https://travis-ci.org/zikula/core/jobs/206548605)

2.18.18 -> [tests fail](https://travis-ci.org/zikula/core/jobs/208185401)

[breaking commit](https://github.com/zikula/core/commit/6b265466a3c10226452200fac0bcf0c4bce3712a) was a simple vendor update

I believe we are correctly `require`-ing the [autoload file](https://github.com/zikula/core/blob/1.4/src/app/autoload.php#L19)

commits: https://github.com/zikula/core/commits/1.4

we have since added a phpunit dependency which wasn't previously included. This solved the `PHP Fatal error:  Class 'PHPUnit\Framework\TestCase' not found...` error, but with that corrected, now two of our **_previously passing_** tests now fail.

both failures are tests of a pseudo-entity type form type
1. https://github.com/zikula/core/blob/1.4/src/system/CategoriesModule/Tests/Form/Type/CategoriesTypeTest.php#L155
2. https://github.com/zikula/core/blob/1.4/src/system/CategoriesModule/Tests/Form/Type/CategoriesTypeTest.php#L219

since nothing else has changed I am very confused what to do and would very much appreciate your help. thanks in advance

**Comment by robfrawley** [comment]
Your issue doesn't seem to have any relationship to the original issue of this ticket __and__ if you want people to see your new issue who weren't explicitly tagged in _this issue_, I'd suggest opening a new one.
